manager/volume.go

```diff
@@ -680,7 +681,31 @@ func (m *VolumeManager) TrimFilesystem(name string) (v *longhorn.Volume, err err
 		return nil, fmt.Errorf("volume frontend is disabled")
 	}
 
-	return v, util.TrimFilesystem(name, v.Spec.Encrypted)
+	if v.Spec.AccessMode == longhorn.AccessModeReadWriteMany {
+		return v, m.trimRWXVolumeFilesystem(name, v.Spec.Encrypted)
+	}
+	return v, m.trimNonRWXVolumeFilesystem(name, v.Spec.Encrypted)
+}
+
+func (m *VolumeManager) trimNonRWXVolumeFilesystem(volumeName string, encryptedDevice bool) error {
+	return util.TrimFilesystem(volumeName, encryptedDevice)
+}
+
+func (m *VolumeManager) trimRWXVolumeFilesystem(volumeName string, isEncryptedDevice bool) error {
```

Comment:
This is not changed yet :slightly_smiling_face: 

`encryptedDevice`

Reply:
Updated. Thank you.